code/obj/rocko.dm

```diff
@@ -0,0 +1,184 @@
+
+// CE's pet rock! A true hellburn companion
+obj/item/rocko
+	name = "Rocko"
+	icon = 'icons/obj/materials.dmi'
+	icon_state = "rock1"
+	w_class = 1
+	force = 10
+	throwforce = 15
+	throw_range = 3
+
+	var/static/list/rocko_is
+	var/smile = TRUE
+	var/painted
+	var/bright = FALSE
+	var/mob/living/holder
+	var/obj/item/clothing/head/hat
+
+	New()
+		. = ..()
+		var/matrix/xf = matrix()
+		if(prob(20))
+			bright = TRUE
+
+		src.chat_text = new
+		src.vis_contents += src.chat_text
+
+		src.icon_state = "rock[pick(1,3)]"
+		src.transform = xf*1.2
+		src.rocko_is = list("a great listener", "a good friend", "trustworthy", "wise", "sweet", "great at parties")
+		src.hat = new /obj/item/clothing/head/helmet/hardhat
```

Comment:
On phone so excuse lack of detail: surely just do:

src.hat = new /the/path(src)

Rather than creating then setting next line?